lib/canvas_editor/events.js

```diff
@@ -70,23 +70,27 @@ function addPointerListeners(canvasElement, drawArea, JavaEditorArea) {
 function addKeyboardListeners(canvasElement, editorArea, JavaEditorArea) {
   const isMac =
     typeof navigator !== 'undefined' && navigator.platform === 'MacIntel';
+  const isMenuKey = (ev) => (isMac && ev.metaKey) || (!isMac && ev.ctrlKey);
 
   function fireKeyEvent(what, ev) {
     const key = getKeyFromEvent(ev, JavaEditorArea);
     if (key === null) return;
-    ev.stopPropagation();
-    ev.preventDefault();
+
     editorArea.fireKeyEvent(
       what,
       key,
       ev.altKey,
       ev.ctrlKey,
       ev.shiftKey,
-      (isMac && ev.metaKey) || (!isMac && ev.ctrlKey),
+      isMenuKey(ev),
     );
   }
 
   canvasElement.addEventListener('keydown', (ev) => {
+    // copy-paste is handled by the clipboard event
+    if (isMenuKey(ev) && ev.key === 'c') return;
+    if (isMenuKey(ev) && ev.key === 'v') return;
```

Comment:
not triggering java events with`fireKeyEvent` in this case, avoid double copy-paste events.
And events triggered from keyboard can't work. Need clipboard events